sapi/fpm/fpm/zlog.c

```diff
@@ -286,14 +286,8 @@ static zlog_bool zlog_stream_buf_alloc_ex(struct zlog_stream *stream, size_t nee
 {
 	char *buf;
 	size_t size = stream->buf.size ?: stream->buf_init_size;
-
-	if (stream->buf.data) {
-		size = MIN(zlog_limit, MAX(size * 2, needed));
-		buf = realloc(stream->buf.data, size);
-	} else {
-		size = MIN(zlog_limit, MAX(size, needed));
-		buf = malloc(size);
-	}
+	size = MIN(zlog_limit, MAX((stream->buf.data ? (size << 1) : size), needed));
+	buf = realloc(stream->buf.data, size);
```

Comment:
I guess this shouldn't be here...

Reply:
already old and fixed yes was wrong branch :)

Reply:
ok can you please create 2 commits not having this change (not adding it in the first commit and removing it in the second one). Please can you also add some sensible commit message so I can just push it... :)